types/src/working-group/index.ts

```diff
@@ -243,22 +240,13 @@ export class WorkingGroupOpeningPolicyCommitment extends JoyStruct<IWorkingGroup
   }
 };
 
-export enum OpeningTypeKeys {
-  Leader = 'Leader',
-  Worker = 'Worker'
-};
 
-export class OpeningType extends Enum {
-  constructor (value?: any, index?: number) {
-    super(
-      {
-        Leader: Null,
-        Worker: Null
-      },
-      value, index
-    );
-  }
-};
+export class OpeningType_Leader extends Null { };
+export class OpeningType_Worker extends Null { };
+export class OpeningType extends JoyEnum({
+  Leader: OpeningType_Leader,
```

Comment:
I prefer this naming style, so perhaps use same for UnslashableTerms, and SlashableTerms. But I wonder if the linter will not like the underscore in a class name..

Reply:
It looks like `@typescript-eslint/camelcase` would disallow it, which is currently a deprecated rule btw (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/camelcase.md).

I was forced to come up with some different names to avoid name conflicts for `Worker` class.
I could also use `OpeningTypeWorker` or, since both classes are just `Null`, abandon the requirement for registering classes for all enum keys and just use `JoyEnum({ Leader: Null, Worker: Null } as const)`,

I think that now that we have a good, TS-compatible `JoyEnum` helper, we don't need to actually use `instanceof`, we can just check `this.type === typeKey` in `isOfType` and `asType`, since:
- `typeKey` is enforced to be one of the registered keys by TypeScript rules
- The same keys are beeing passed to `Enum` constuctor (`@polkadot/api`) and become available in `this._def` object, which is then used in `this.type` getter
So it should be a pretty type-safe and error-resistant way still.